Dockerfile

```diff
@@ -55,43 +55,26 @@ RUN set -ex \
     json-c-devel \
     libyaml-devel \
     libjwt-devel \
+    python3 \
     && yum clean all \
     && rm -rf /var/cache/yum
 
-# Set Vim and Git defaults
-RUN set -ex \
-    && echo "syntax on"           >> "$HOME/.vimrc" \
-    && echo "set tabstop=4"       >> "$HOME/.vimrc" \
-    && echo "set softtabstop=4"   >> "$HOME/.vimrc" \
-    && echo "set shiftwidth=4"    >> "$HOME/.vimrc" \
-    && echo "set expandtab"       >> "$HOME/.vimrc" \
-    && echo "set autoindent"      >> "$HOME/.vimrc" \
-    && echo "set fileformat=unix" >> "$HOME/.vimrc" \
-    && echo "set encoding=utf-8"  >> "$HOME/.vimrc" \
-    && git config --global color.ui auto \
-    && git config --global push.default simple
-
 # Add Tini
 ENV TINI_VERSION v0.18.0
 ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
 RUN chmod +x /tini
 
-# Install Python versions
-ARG PYTHON_VERSIONS="3.6 3.7 3.8 3.9"
-COPY files/install-python.sh /tmp
-RUN set -ex \
-    && for version in ${PYTHON_VERSIONS}; do /tmp/install-python.sh "$version"; done \
-    && rm -f /tmp/install-python.sh
+FROM base as builder
 
 # Compile, build and install Slurm from Git source
 ARG SLURM_TAG=slurm-21-08-0-1
 RUN set -ex \
     && git clone -b ${SLURM_TAG} --single-branch --depth=1 https://github.com/SchedMD/slurm.git \
     && pushd slurm \
-    && ./configure --prefix=/usr --sysconfdir=/etc/slurm \
+    && ./configure --quiet --disable-dependency-tracking --prefix=/usr --sysconfdir=/etc/slurm --enable-load-env-no-login \
```

Comment:
What do these options do?

Reply:
Some relevant parts of Slurm repo I could find

`--enable-load-env-no-login`:
[x_ac_env.m4](https://github.com/SchedMD/slurm/blob/4801c60da4784346c4bf830a0b198364012e44ee/auxdir/x_ac_env.m4#L18)
[config.h.in](https://github.com/SchedMD/slurm/blob/1ca5f08f038c389b5073ea3f00de3b4aef628467/config.h.in#L416)
[env.c](https://github.com/SchedMD/slurm/blob/1225a45876c055b75654e2817571bb8393fc414b/src/common/env.c#L2065)

So I got the impression that this allowed loading of user environments with a `.login` file being present. I thought this may have been the source of issues because bash doesn't use this dotfile, and centos doesn't have it in user directories by default. I don't think adding it really affected anything though. Very hard to trial and error because of the need to do a full rebuild.

`--disable-dependency-tracking`:
[configure](https://github.com/SchedMD/slurm/blob/f4bfd747434a92715fa1e5f996ea498532450b0a/configure#L1693)

So I got the impression this would speed up the build in a docker build - but not quite sure what the implications are.

If you think best to remove, I'm happy with that. Will attempt to rebuild and check the affects!

Reply:
Based on the `env.c` this is the different it seems to make:

https://superuser.com/questions/453988/whats-the-difference-between-su-with-and-without-hyphen

Not sure what implications this has for the rest of Slurm - if it isn't required to make the environment work maybe best to leave it out as we don't fully understand what it does.

Reply:
Ah yeah I thought the same - though I think I misinterpreted which one brought the calling environment in. It does seem to be we want to behavior **with** the hyphen, so I've remove the flag now.

Interestingly - this does seem to be an issue/bug with slurm. The `--export=NONE` is needed to stop environment variables coming through from the calling environment.

Reply:
Also removed the depency tracking flag